toolchain/driver/driver.cpp

```diff
@@ -30,6 +30,45 @@
 
 namespace Carbon {
 
+auto Driver::FindPreludeFiles(llvm::StringRef data_dir,
+                              llvm::raw_ostream& error_stream)
+    -> llvm::SmallVector<std::string> {
+  llvm::SmallVector<std::string> result;
+
+  // Include <data>/core/prelude.carbon, which is the entry point into the
+  // prelude.
+  {
+    llvm::SmallString<256> prelude_file(data_dir);
+    llvm::sys::path::append(prelude_file, llvm::sys::path::Style::posix,
+                            "core/prelude.carbon");
+    result.push_back(prelude_file.str().str());
+  }
+
+  // Glob for <data>/core/prelude/**/*.carbon and add all the files we find.
+  llvm::SmallString<256> prelude_dir(data_dir);
+  llvm::sys::path::append(prelude_dir, llvm::sys::path::Style::posix,
+                          "core/prelude");
+  std::error_code ec;
+  for (llvm::sys::fs::recursive_directory_iterator prelude_files_it(
+           prelude_dir, ec, /*follow_symlinks=*/false);
+       prelude_files_it != llvm::sys::fs::recursive_directory_iterator();
+       prelude_files_it.increment(ec)) {
+    if (ec) {
+      error_stream << "ERROR: Could not find prelude: " << ec.message() << "\n";
+      result.clear();
+      break;
+    }
+
+    // TODO: Should we ignore case?
```

Comment:
Can you elaborate on this todo? What's the situation where this function would encounter file extensions with other casing choices? Note, we already have other file extension checking code: I would not expect case-insensitive logic here to be sufficient.

Reply:
OK, if we're already doing a case-sensitive extension comparison, let's just drop the TODO. I think it's fine to insist that the carbon installation preserves the lowercase extensions (we're already using a more-than-three-character extension, so it's not like we need to worry about FAT filenames anyway, and everyone else I think we might care about is case-preserving).

```suggestion
```